src/core/queue/__tests__/event-queue.test.ts

```diff
@@ -11,16 +11,20 @@ import { Plugin } from '../../plugin'
 import { EventQueue } from '../event-queue'
 
 async function flushAll(eq: EventQueue): Promise<Context[]> {
-  let flushed: Context[] = []
-
+  const flushSpy = jest.spyOn(eq, 'flush')
   await pWhile(
     () => eq.queue.length > 0,
     async () => {
-      const res = await eq.flush().catch(() => [])
-      flushed = flushed.concat(res)
+      return new Promise((r) => setTimeout(r, 0))
```

Comment:
I had to change this test code because the previous implementation was relying on `scheduleFlush` to be called initially with a timeout of 500 ms, and that the `flush` calls invoked would complete in less than 500 ms.

Since I removed the initial `scheduleFlush` from the EventQueue constructor, the 1st time it's invoked is with a timeout of 0 ms, so we weren't capturing all the meaningful `flush` calls.